src/core/db/backend/pg.py

```diff
@@ -764,3 +768,158 @@ def import_file_w_dbtruck(self, table_name, file_path):
 
         return import_datafiles([file_path], create_new, table_name, errfile,
                                 PGMethods, **dbsettings)
+
+    def escape_quotes(self, parameter):
+        return parameter.replace("'", "''")
+
+    def check_access_permissions(self, grantor, repo_base):
+        '''
+        Checks if the repo owner is the person granting security policies.
+        Only repo owners can create security policies.
+        '''
+        if grantor == repo_base:
+            return True
+        return False
+
+    def create_security_policy(self, policy, policy_type, grantee, grantor,
+                               table, repo, repo_base):
+        '''
+        Creates a new security policy in the policy table if the policy
+        does not yet exist.
+        '''
+        params = [policy, policy_type, grantee, grantor,
+                  table, repo, repo_base]
+        params = [self.escape_quotes(param) for param in params]
+        for param in params[2:]:
+            self._check_for_injections(param)
+
+        if not self.check_access_permissions(grantor, repo_base):
+            raise Exception('%s does not have permission to define security '
+                            'policies on %s.%s.' % (grantor, repo, table))
+
+        # Raise an exception if the security policy already exists in the table
+        security_policy = self.find_security_policy(
+            table, repo, repo_base, policy=policy, policy_type=policy_type, grantee=grantee,
+            grantor=grantor)
+        if security_policy != []:
+            raise Exception('Security policy already exists in table.')
```

Comment:
e.g. If you raise PermissionDenied() the user will receive a 403 Forbidden response with the exception's message on the page, ValueError() for 400 Bad Request (user error), LookupError() for 404 Not Found, and most everything else result in a a 500 Internal Server Error.